File: OnlineShopAPI/PaymentInfoMicroservice/PaymentInfo.Api/Models/UpdatePaymentRequest.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Threading.Tasks;

namespace PaymentInfo.Api.Models
{
    public class UpdatePaymentRequest
    {
        public long PaymentId { get; set; }

        public string NewNameOnCard { get; set; }

        public string NewCardNumber { get; set; }

        public string NewSecurityCode { get; set; }

        public string NewExpDate { get; set; }

        public long NewCardTypeId { get; set; }
    }
}

File: OnlineShopUI/OnlineShop/src/app/model/requests/update-payment-request.ts
export class UpdatePaymentRequest {
    paymentId: number;
    newNameOnCard: string;
    newCardNumber: string;
    newSecurityCode: string;
    newExpDate: string;
    newCardTypeId: number;

    constructor(paymentId: number, newNameOnCard: string, newCardNumber: string, newSecurityCode: string, newExpDate: string, newCardTypeId: number){
        this.paymentId = paymentId;
        this.newNameOnCard = newNameOnCard;
        this.newCardNumber = newCardNumber;
        this.newSecurityCode = newSecurityCode;
        this.newExpDate = newExpDate;
        this.newCardTypeId = newCardTypeId;
    }
}
